File: C++/A/400A-innaandchoose.cpp
#include <bits/stdc++.h>
using namespace std;
void func()
{
    string s;
    cin >> s;
    int a[] = {1, 2, 3, 4, 6, 12};
    int count = 0;
    vector<int> v;
    for (int i = 0; i < 6; i++)
    {
        int rows = a[i], cols = 12 / a[i], ct = 0;
        for (int j = 0; (j < cols) && (ct == 0); j++)
        {
            int temp = 0;
            for (int k = 0; k < rows; k++)
                temp += (s[j + k * cols] == 'X');

            if (temp == rows)
                ct = 1;
        }
        if (ct == 1)
            v.push_back(a[i]);
        count += ct;
    }
    cout << count << " ";
    for (int i = 0; i < v.size(); i++)
        cout << v[i] << "x" << 12 / v[i] << " ";
    cout << '\n';
    return;
}

int main()
{
    int t;
    cin >> t;
    while (t--)
        func();
}

File: Python/A/400A-innaandchoose.py
def func():
    s = input().strip()
    a = [1, 2, 3, 4, 6, 12]
    count = 0
    v = []
    for i in range(6):
        rows, cols, ct = a[i], 12 // a[i], 0
        for j in range(cols):
            temp = sum(1 for k in range(rows) if s[j + k * cols] == 'X')
            if temp == rows:
                ct = 1
        if ct == 1:
            v.append(a[i])
        count += ct

    print(count, end=" ")
    for i in range(len(v)):
        print(f"{v[i]}x{12 // v[i]}", end=" ")
    print()


t = int(input())
while t > 0:
    func()
    t -= 1
